### PR TITLE
Update to 1.2.14

### PR DESCRIPTION
**Pull request by bbhtt**
I wasn't sure what the master.mint and master.lmde tags were for, so I choose the last versioned tag. Please let me know if I need to use those instead.

I added `commit` to all git sources to protect against retagging.

### CODE REVIEW

**Comment by bbhtt** [comment]
> Started [test build 109669](https://buildbot.flathub.org/#/builders/32/builds/109669)

`ERROR: Could not find a version that satisfies the requirement setuptools>=62.4 (from setuptools-rust) (from versions: none)
ERROR: No matching distribution found for setuptools>=62.4`

Runtime has older setuptools than what's needed by setuptools-rust 1.5.1.

**Comment by bbhtt** [comment]
bot, build

**Comment by bbhtt** [comment]
bot, build

**Comment by bbhtt** [comment]
> Build [109684 failed](https://buildbot.flathub.org/#/builders/32/builds/109684)

It built fine but looks like Flathub is having issues

`Api call to https://hub.flathub.org/api/v1/build/107320/upload failed with status 408, details: {'status': 408, 'error-type': 'no-error', 'message': 'No json error details from server'}`

bot, build

File: python3-modules.json [hunk 1]
```diff
@@ -12,8 +12,8 @@
             "sources": [
                 {
                     "type": "file",
-                    "url": "https://files.pythonhosted.org/packages/4b/ad/8aa7cc8e1e98aee745a1f6a58a6dc3534e806b3d7d3b4e6d80b8f51d7ecc/grpcio-1.47.0.tar.gz",
-                    "sha256": "5dbba95fab9b35957b4977b8904fc1fa56b302f9051eff4d7716ebb0c087f801"
+                    "url": "https://files.pythonhosted.org/packages/e2/13/0f91c6c7eb0d934141743c7663e8808f4dc478dd0a7b0fd2a02a286c0d6d/grpcio-1.48.1.tar.gz",
```
Comment (mtwebster) [inline python3-modules.json]: We actually need to go back down to 1.44, as anything higher than that is running with very high cpu use - see https://github.com/grpc/grpc/issues/30428#issuecomment-1234519274

Comment (bbhtt) [reply python3-modules.json]: Thanks!

**Comment by mtwebster** [comment]
Just out of curiousity, how should I know which version of rust-stable to install? I tried the latest first - 22.08, and the build still couldn't find it. Installing 21.08 worked.

I hardly ever work with flatpaks (even less now with you on the job :)), so maybe I'm missing something.

Thanks again

**Comment by bbhtt** [comment]
> Just out of curiousity, how should I know which version of rust-stable to install? I tried the latest first - 22.08, and the build still couldn't find it. Installing 21.08 worked.

Yea, the error there is a bit confusing. It has to match with the version of the base SDK/runtime. Warpinator uses GNOME 42 which is based on FDO-21.08, so you can only use a sdk-extension based on that same version.

To use the 22.08 branch of rust-stable warpinator needs to use GNOME 43 which is based on FDO-22.08. It is yet to be released on Flathub.

**Comment by mtwebster** [comment]
Ah ok - you'd think the builder might be a bit more helpful about it - or at least `flatpak install` could be - I didn't have Gnome 43 Sdk installed.

Thanks for the explanation.

**Pull request merged**